File: 0_1.baekjoon/two-pointer/java/BOJ1484.java
import java.io.*;
import java.util.*;

class BOJ1484 {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int G = Integer.parseInt(br.readLine());
        
        int left = 1, right = 1;
        List<Integer> result = new ArrayList<>();

        while ((int)(Math.pow(right, 2) - Math.pow(right - 1, 2)) <= G) {
            int diff = (int)(Math.pow(right, 2) - Math.pow(left, 2));

            if (diff == G) {
                result.add(right++);
            } else if (diff > G) {
                left++;
            } else {
                right++;
            }
        }

        if (result.isEmpty()) {
            System.out.println(-1);
        } {
            result.forEach(r -> System.out.println(r));
        }
    }
}

File: 0_1.baekjoon/two-pointer/python/BOJ1484.py
from sys import stdin
readline = stdin.readline

G = int(readline().strip())

def solve():

    left, right, result = 1, 1, []

    while right ** 2 - (right - 1) ** 2 <= G:
        diff = right ** 2 - left ** 2

        if diff == G:
            result.append(right)
            right += 1

        elif diff > G:
            left += 1
        
        else:
            right += 1

    if not result:
        print(-1)
    else:
        for num in result:
            print(num)

solve()

